roles/content_view_promotion_rollback_publish/tasks/promote.yml

```diff
@@ -0,0 +1,82 @@
+---
+# get data on the current content-view
+- name: "Gather Data For Current Content-View From Foreman"
+  theforeman.foreman.resource_info:
+    username: "{{ foreman.user }}"
+    password: "{{ foreman.password }}"
+    server_url: "{{ foreman.url }}"
+    organization: "{{ organization.key }}"
+    resource: content_views
+    search: name = "{{ content_view }}"
+    validate_certs: false
+  register: content_view_data
```

Comment:
You can use the dedicated info module now: https://github.com/theforeman/foreman-ansible-modules/blob/develop/plugins/modules/content_view_info.py

Reply:
Does it make sense to have roles that utilize modules that aren't even in a main version yet?  By using the existing format, this role will be backwards compatible with older versions of the collection.

Reply:
Chiming in here, I think it does make sense. The primary way we expect users would consume this role would be to install the collection.

If they install the development branch from git they would have access to all of the latest modules. Of course it is more likely that they install one of the released versions from ansible galaxy, or a downstream RPM packaged version.... in each case still by the time this role is included in any of those releases, so too would be all modules currently present at development time.

Is this contrary to your expectation, @bmarlow ?

Reply:
@wbclark I think its fair to assume people are going to get this by way of the collection so no need to worry there.

I guess my statement is more that I don't think it makes sense to build a role against a module that is in development as changes that happen to that module could potentially break the role.

As it sits (and based on the way that roles are consumed/delivered), I would think we wouldn't care which module got used as long as it worked (unless there was a significant performance benefit or depreciation issue) due to the fact that the user shouldn't really be modifying the role, but rather changing the variables that the role digests.

I haven't had a chance to use the new module yet, however my biggest concern would be that the data returned isn't structured the same way (maybe it is?) and it would require more significant refactoring (and lets face it, the data returned isn't the most easily parsable.